kgforge/specializations/databases/store_database.py

```diff
@@ -0,0 +1,97 @@
+#
+# Blue Brain Nexus Forge is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Blue Brain Nexus Forge is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Blue Brain Nexus Forge. If not, see <https://choosealicense.com/licenses/lgpl-3.0/>.
+from typing import Callable, Optional, Any
+
+from kgforge.core.archetypes import Store, Database
+from kgforge.core.commons.execution import not_supported
+from kgforge.specializations.stores.bluebrain_nexus import BlueBrainNexus
+from kgforge.specializations.databases.utils import type_from_filters
+
+
+class StoreDatabase(Database):
+    """A high-level class to retrieve and create Database-related Resources."""
+    
+
+    _REQUIRED = ("name", "origin", "source", "model")
+
+    def __init__(self, forge: Optional["KnowledgeGraphForge"],
+                 **config) -> None:
+        """
+        The properties defining the StoreDatabase are:
+
+        :param forge: To use forge utilities 
+           name: <the name of the database> - REQUIRED
+           origin: <'store'> - REQUIRED
+           source: <a directory path, an URL, or the class name of a Store> - REQUIRED
+           bucket: <when 'origin' is 'store', a Store bucket>
+           endpoint: <when 'origin' is 'store', a Store endpoint>
+           token: <when 'origin' is 'store', a Store token
+           model:
+             origin: <'directory', 'url', or 'store'>
+             source: <a directory path, an URL, or the class name of a Store>
+             context: <'directory', 'url', or 'store'>
+                bucket: <when 'origin' is 'store', a Store bucket>
+                iri: <the IRI of the origin>
+        """
+        self._check_properties(**config)
+        self.name = config.pop('name')
+        source = config.pop('source')
+        super().__init__(forge, source, **config)
+
+    def _check_properties(self, **info):
+        properties = info.keys()
+        for r in self._REQUIRED:
+            if r not in properties:
+                raise ValueError(f'Missing {r} from the properties to define the DatabasSource')
```

Comment:
Little typo (Databas). 
Also if multiples properties are missing, ValueError would fail only for one and then after fixing, would fail for the next. You could check all missing properties first, and then raise the Error with an array of missing properties?